### PR TITLE
Use more datatypes

### PR DESCRIPTION
**Pull request by ekohl**
Perhaps there are more fields that can use better data types.

### CODE REVIEW

File: manifests/config/passenger.pp [hunk 1]
```diff
@@ -68,8 +68,8 @@
   String $priority = $::foreman::vhost_priority,
   String $servername = $::foreman::servername,
```
[the remark this answers is not in the document]
Comment (ekohl) [reply manifests/config/passenger.pp]: See, I said there are probably more fields :) Good one.

**Comment by ekohl** [comment]
Also changed the `server_ssl_certs_dir` and `server_ssl_crl` to be `Variant[Enum[''], Stdlib::Absolutepath]`

**Comment by mmoll** [comment]
@ekohl What's the status here? Still WIP?

**Comment by ekohl** [comment]
No, I think this can be merged. Even if I missed a few, I think this can be reviewed to be merged.

**Comment by mmoll** [comment]
merged, bedankt @ekohl!

**Pull request merged**